### ISSUE TITLE
Why RandomResize for evaluation?

### ISSUE THREAD

**Issue by jackkwok** [facebookresearch/detr#161]
Hi, in coco.py, the validation set transform includes a resize transform:

https://github.com/facebookresearch/detr/blob/master/datasets/coco.py#L140

Why was 800 chosen?  If the authors can shed some light, that will be helpful.  In my custom dataset where images are generally small (200x200), choosing a smaller resize number (400) improved the mAP metrics.  I also tried removing the RandomResize line but that hurt the metrics number quite a bit which I cannot explain.

**Comment by alcinos** [comment]
Hi @jackkwok 
Thanks for your interest in DETR

800 is a very common value for COCO, we followed standard practice there. See for example Detectron2: https://github.com/facebookresearch/detectron2/blob/45a4d970bb2242ce0630c5ce03f860ad14382169/detectron2/config/defaults.py#L56-L57

If you have very small images, it may be the case that upsampling to 800 creates artifacts that affect performance.

If you remove the RandomResize, two things will occur, both of which will hurt your final AP:

1. Image sizes will be quite significantly out of distribution, since the model is not trained on such small images.
2. Objects will be relatively smaller. This is a potential issue, especially if the objects you want to detect are relatively small. The current version of DETR is known to suffer a bit on small objects, so in general making objects even smaller will decrease performance.

I believe I have answered your question, and as such I'm closing this. Feel free to reach out if you have further concerns.

**Comment by jackkwok** [comment]
Thanks @alcinos .  Do you have recommendation on what resize scales to use for the training set transform when images are mostly small (200x200)?

**Comment by alcinos** [comment]
I don't think there is a generic recommendation that could fit all cases. Try different values and see what gives you the best AP

**Comment by fmassa** [comment]
@netw0rkf10w is right: there is no randomness in the evaluation, and `T.RandomResize([800], max_size=1333)` is equivalent to `T.Resize(800)` with the size limit
